internal/agent/client/compose.go

```diff
@@ -17,11 +17,12 @@ import (
 const defaultPodmanTimeout = 2 * time.Minute
 
 type ComposeSpec struct {
-	Services map[string]ComposeService `yaml:"services"`
+	Services map[string]ComposeService `json:"services"`
 }
 
 type ComposeService struct {
-	Image string `yaml:"image"`
+	Image         string `json:"image"`
+	ContainerName string `json:"container_name"`
```

Comment:
note: serialization would fail using yaml but works fine using json which the module supports both.